manifests/install/windows.pp

```diff
@@ -49,6 +49,25 @@
     ensure  => file,
     content => file('puppet_agent/install_puppet.ps1')
   }
+
+  $_puppet_check = windows_native_path("${::env_temp_variable}/puppet_check.ps1")
+  file { $_puppet_check:
+    ensure  => file,
+    content => file('puppet_agent/puppet_check.ps1')
+  }
+
+  exec { 'puppet_check.ps1':
+    command => "${::system32}\\WindowsPowerShell\\v1.0\\powershell.exe \
+                  -NoProfile \
+                  -NoLogo \
+                  -NonInteractive \
+                  ${_puppet_check} ${::puppet_agent::_expected_package_version} ${_msi_location}",
+    returns => 0,
+    require => [
+      File[$_puppet_check]
+    ]
+  }
```

Comment:
I added this so that puppet runs will fail in case the powershell script does not return 0.
Without this check Puppet Enterprise will not detect any problems if the `onlyif` from `exec { 'install_puppet.ps1':` will fail.